actions/action.go

```diff
@@ -52,7 +52,7 @@ func NewOutputs(uri string, latestVersion *semver.Version, additionalOutputs Out
 	outputs := Outputs{
 		"sha256":  sha256,
 		"uri":     uri,
-		"version": fmt.Sprintf("%d.%d.%d", latestVersion.Major(), latestVersion.Minor(), latestVersion.Patch()),
+		"version": latestVersion.String(),
```

Comment:
this may have knock on effects, but there are no tests to say why the code was doing this in the first place.

Reply:
>Tomee v9 is pre-release only, I've had to change how the version is outputted to use the full version rather than major.minor.patch - I'm not entirely sure how pre-releases were working before this?

There is a core assumption baked into not just the actions but libpak and its dependency management that versions will be valid semver (at least in terms of structure). The actions that update dependencies take on the responsibility of enforcing this. If the particular dependency the action is updating does not use semver then it needs to adjust the version somehow to force it into a semver.

[Stackdriver](https://github.com/paketo-buildpacks/pipeline-builder/blob/main/actions/google-stackdriver-profiler-dependency/main.go#L54) is an example of this as they use a date-based version.

I'm kind of surprised Tomcat was doing this as their version numbers should fit into semver.